cola/widgets/remote.py

```diff
@@ -698,32 +700,33 @@ def push_to_all(self, _remote, *args, **kwargs):
 
             all_results = combine(result, all_results)
 
-        try:
-            notification = Notify()
+        if prefs.notify_on_push(self.context):
+            try:
+                notification = Notify()
 
-            total_remotes = len(selected_remotes)
+                total_remotes = len(selected_remotes)
 
-            if pushed_remotes == total_remotes:
-                notification.title = N_('Push Successful')
-                notification.icon = resources.package_data('icons', 'git-cola.svg')
-            elif pushed_remotes != 0:
-                notification.title = N_('Push Partially Successful')
-                notification.icon = resources.package_data(
-                    'icons', 'git-cola-error.svg'
-                )
-            else:
-                notification.title = N_('Push Failed')
-                notification.icon = resources.package_data(
-                    'icons', 'git-cola-error.svg'
-                )
+                if pushed_remotes == total_remotes:
+                    notification.title = N_('Push Successful')
+                    notification.icon = resources.package_data('icons', 'git-cola.svg')
+                elif pushed_remotes != 0:
+                    notification.title = N_('Push Partially Successful')
+                    notification.icon = resources.package_data(
+                        'icons', 'git-cola-error.svg'
+                    )
+                else:
+                    notification.title = N_('Push Failed')
+                    notification.icon = resources.package_data(
+                        'icons', 'git-cola-error.svg'
+                    )
 
-            notification.message = N_(
-                f'{pushed_remotes}/{total_remotes} remotes updated'
-            )
+                notification.message = N_(
+                    f'{pushed_remotes}/{total_remotes} remotes updated'
+                )
 
-            notification.send()
-        except (ImportError, ModuleNotFoundError):
-            log.show_remote_messages(self, N_('notifypy dependency not found'))
+                notification.send()
+            except (ImportError, ModuleNotFoundError):
+                log.show_remote_messages(self, N_('notifypy dependency not found'))
```

Comment:
This is too much code under the try/except block. Also, this importError check should be at the top of the file when we try to import from notifypy.

If we hit the ImportError we should set Notify to None.

Then, move all of this logic into a standalone private helper function. Then this part can just have a single line where it calls that function.

That new function, let's call it `def _emit_push_notification(...)`, can check `if Notify is None:` and `return` early.